src/main.rs

```diff
@@ -55,31 +58,51 @@ fn main() {
     }
 }
 
-#[tokio::main]
-async fn start(cli: Cli) -> Result<i32, Box<dyn std::error::Error>> {
-    let cron = cli.cron();
-    let log = cli.log();
-
-    let tasks = TaskTracker::new();
+fn spawn_child(
+    cli: &Cli,
+    tasks: &TaskTracker,
+) -> io::Result<(
+    Child,
+    Option<UnboundedReceiver<String>>,
+    Option<UnboundedReceiver<String>>,
+)> {
+    let should_stdout = cli.should_pipe_stdout();
+    let should_stderr = cli.should_pipe_stderr();
 
-    let mut child = command(&cli.command, &log).spawn()?;
+    let mut child = command(&cli.command, should_stdout, should_stderr).spawn()?;
```

Comment:
I noticed I didn't get an error when wrap couldn't start a process because it wasn't executable `chmod -x test.sh`. Would be nice if it could report that start error as an error as well.